### PR TITLE
Replaced "components" with "coordinates" and altered wording to suit

### PR DESCRIPTION
**Pull request by jonblower**
… following discussion on issue #77

### CODE REVIEW

File: spec.md [hunk 1]
```diff
@@ -496,12 +496,12 @@ Its general structure is:
...
+- The value of `"dataType"` determines the structure of an axis value and its coordinates that are made available for referencing. The values of `"dataType"` defined in this specification are `"primitive"`, `"tuple"`, and `"polygon"`. Custom values MAY be used as detailed in the "Extensions" section. For `"primitive"`, there is a single coordinate identifier and each axis value MUST be a number or string. For `"tuple"`, each axis value MUST be an array of fixed size of primitive values in a defined order, where the tuple size corresponds to the number of coordinate identifiers. For `"polygon"`, each axis value MUST be a GeoJSON Polygon coordinate array, where the order of coordinates in each of the coordinate pairs is given by the `"coordinates"` array.
```
Comment (letmaik) [inline spec.md]: "coordinate pairs" is too restrictive, as the polygon can have more than 2 coordinates in each point. we restrict that in the common domain types, but not in the data type.

Comment (jonblower) [inline spec.md]: OK, how about (simpler actually):

For `"polygon"`, each axis value MUST be a GeoJSON Polygon coordinate array, where the order of coordinates is given by the `"coordinates"` array.

Comment (letmaik) [inline spec.md]: Yes, sounds fine.

**Comment by letmaik** [comment]
With the polygon fix, I think this may be fine, but before merging this, could you run this against somebody to see if it makes sense? We changed this particular field several times already, so this should be the final change.

The only worry I have is that the new name may be a bit inconsistent (see NdArray's "axisNames" which is not "axes"). So, technically, it should be "coordinateNames". This would also avoid the overload with GeoJSON's "coordinates" member which has the actual coordinates. Having said that, I find "coordinateNames" too unhandy.

**Comment by jonblower** [comment]
Yes, I can run this past others to see if it makes sense. Assuming it's OK, should I edit the branch and then make another pull request, or merge them and then make the change?

I would think that "axisNames" is still fine. In NdArray, these really are the names of the axes, not the coordinates. For primitive axes they are the same thing, but for tuples and polygons you would still use the axis name (e.g. `"composite"`) in the NdArray. Or have I missed your point?

**Comment by letmaik** [comment]
> Yes, I can run this past others to see if it makes sense. Assuming 
> it's OK, should I edit the branch and then make another pull request, 
> or merge them and then make the change?
>
Just add another commit to this branch, and then we can merge that. You 
can squash commits when merging via github.

> I would think that "axisNames" is still fine. In NdArray, these really 
> are the names of the axes, not the coordinates. For primitive axes 
> they are the same thing, but for tuples and polygons you would still 
> use the axis name (e.g. |"composite"|) in the NdArray. Or have I 
> missed your point?
>
I just meant the suffix "Names", as the "coordinates" are in the end 
also names, and not actual coordinates. Or identifiers, as they are now 
called in the spec.

**Comment by jonblower** [comment]
Ah, I see what you mean. Yes, I supposed "coordinateNames" would be technically more consistent but I think this is less friendly than "coordinates". (And "coordinateIdentifiers" would be worse I think!)

**Comment by jonblower** [comment]
Actually, is there a reason we need to call it "axisNames" and not just "axes" in the NdArray? The latter would be more consistent with "coordinates". But I guess it might not be good to use "axes" for one thing in the Domain and another thing in the NdArray, maybe that's why they are different. (I have a feeling we've discussed this before but I can't find the conversation now.)

**Comment by letmaik** [comment]
Yes, overloading is not a good idea, especially since you can't do that 
with a single JSON-LD context (ignoring the fact for now that we don't 
use ld for range data).

**Comment by jonblower** [comment]
I wonder if `"domainAxes"` might be better than `"axisNames"`? (I have a feeling this might have been suggested before too.) This gives the user an explicit clue that the identifiers come from the Domain, whilst also avoiding the use of the ambiguous term "names".

**Comment by letmaik** [comment]
This would couple it. NdArray should be a general purpose thing.

**Comment by jonblower** [comment]
But the concept of "axes" has to be defined _somewhere_, so whether you use "axisNames" or "domainAxes" there is inevitable coupling with something, unless we think that "axis" is a completely general concept that applies to standalone nD arrays. If we wanted to completely decouple the NdArray from the domain, we wouldn't have "axisNames" either. We'd have a separate domain and range and a third object that links the two, but this would be far too awkward.

Personally I think that "domainAxes" is fine, as the concept of a "domain" is common, not just a CovJSON thing. But I don't feel strongly and I'm happy to keep "axisNames". I guess the NdArray is a "labelled but not georeferenced" array.

**Comment by letmaik** [comment]
On 03/08/2016 10:49, Jon Blower wrote:
>
> But the concept of "axes" has to be defined /somewhere/, so whether 
> you use "axisNames" or "domainAxes" there is inevitable coupling with 
> something, unless we think that "axis" is a completely general concept 
> that applies to standalone nD arrays.
>
Yes, it's completely general and decoupled from the domain or coverage.

> If we wanted to completely decouple the NdArray from the domain, we 
> wouldn't have "axisNames" either.
>
Why not? Axis names are just labels for axes, nothing more. The coverage 
object establishes the link from domain axes to range axes, but putting 
a constraint on the NdArray saying that the axis names must match the 
domain names, but this is not defined in the NdArray specification. See 
https://covjson.org/spec/#ndarray-objects for example.

**Comment by jonblower** [comment]
Fair enough. But I couldn't find the place in the spec that says that the axisNames must match the domain axis identifiers (but not necessarily in the same order of course). Did I miss it?

**Comment by letmaik** [comment]
It's at 6.4: https://covjson.org/spec/#coverage-objects

**Comment by jonblower** [comment]
Ah right yes. I did a search on "axisNames" but that particular word isn't in that section. Doesn't matter.

**Comment by jonblower** [comment]
All team members seem to be happy with this change. Shall we go ahead?

**Comment by letmaik** [comment]
OK, please also change the examples in the domain-types.md file. Then we can merge it.

**Comment by jonblower** [comment]
OK done. Do you want to do the merge or shall I?

**Comment by letmaik** [comment]
Let me do it in the evening, then I'll also update the libraries and 
website all in one go, more or less.

**Pull request merged**